Detectors/MUON/MID/Workflow/src/tracks-reader-workflow.cxx

```diff
@@ -13,114 +13,22 @@
-#include "Framework/Task.h"
-
-#include "DPLUtils/RootTreeReader.h"
-
-#include "DataFormatsMID/ROFRecord.h"
-#include "DataFormatsMID/Track.h"
+#include <vector>
+#include "Framework/ConfigParamSpec.h"
+#include "MIDWorkflow/TrackReaderSpec.h"
 
 using namespace o2::framework;
-using namespace o2::mid;
 
-class TrackSamplerTask
+// we need to add workflow options before including Framework/runDataProcessing
+void customize(std::vector<ConfigParamSpec>& workflowOptions)
 {
- public:
-  /// prepare the reader
-  void init(InitContext& ic)
-  {
-    auto inputFileName = ic.options().get<std::string>("infile");
-    mMinNumberOfROFsPerTF = ic.options().get<int>("repack-rofs");
-
-    mReader = std::make_unique<RootTreeReader>("midreco", inputFileName.c_str(), -1,
-                                               RootTreeReader::PublishingMode::Single,
-                                               RootTreeReader::BranchDefinition<std::vector<Track>>{
-                                                 Output{"MID", "TRACKS", 0, Lifetime::Timeframe}, "MIDTrack"},
-                                               RootTreeReader::BranchDefinition<std::vector<ROFRecord>>{
-                                                 Output{"MID", "TRACKSROF", 0, Lifetime::Timeframe}, "MIDTrackROF"},
-                                               &mAccumulator);
-  }
-
-  /// process the next entry
-  void run(ProcessingContext& pc)
-  {
-    if (mReader->next()) {
-      (*mReader)(pc);
-      if (mROFs.size() >= mMinNumberOfROFsPerTF) {
-        publish(pc.outputs());
-      }
-    } else {
-      if (mROFs.size() > 0) {
-        publish(pc.outputs());
-      }
-      pc.services().get<ControlService>().endOfStream();
-    }
-  }
-
- private:
-  /// accumulate the data
-  bool accumulate(std::string_view name, char* data)
-  {
-    if (name == "MIDTrackROF") {
-
-      auto rofs = reinterpret_cast<std::vector<ROFRecord>*>(data);
-
-      // accumulate the ROFs, shifting the track indexing accordingly
-      size_t offset = (mROFs.size() > 0) ? mROFs.back().firstEntry + mROFs.back().nEntries : 0;
-      std::transform(rofs->begin(), rofs->end(), std::back_inserter(mROFs), [offset](const ROFRecord& rof) {
-        return ROFRecord{rof, rof.firstEntry + offset, rof.nEntries};
-      });
-
-    } else if (name == "MIDTrack") {
-
-      auto tracks = reinterpret_cast<std::vector<Track>*>(data);
-
-      // accumulate the tracks
-      mTracks.insert(mTracks.end(), tracks->begin(), tracks->end());
-
-    } else {
-      throw std::invalid_argument("invalid branch");
-    }
-
-    return true;
-  }
-
-  /// publish the data and clear the internal vector
-  void publish(DataAllocator& out)
-  {
-    out.snapshot(OutputRef{"rofs"}, mROFs);
-    out.snapshot(OutputRef{"tracks"}, mTracks);
-    mROFs.clear();
-    mTracks.clear();
-  }
+  workflowOptions.emplace_back("disable-mc", VariantType::Bool, true, ConfigParamSpec::HelpString{"Disable MC info"});
```

Comment:
the standard policy is to have `"disable-mc", VariantType::Bool, false`. Then, in the raw data processing it is set to true via option from env.var common for all workflows.